classes/local/store/s3/client.php

```diff
@@ -390,6 +395,19 @@ public function support_presigned_urls() {
      * @return string.
      */
     public function generate_presigned_url($contenthash, $headers) {
+        $contentdisposition = $this->get_header($headers, 'Content-Disposition');
+        if ($contentdisposition !== '') {
+            $params['ResponseContentDisposition'] = $contentdisposition;
+        }
+
+        $contenttype = $this->get_header($headers, 'Content-Type');
+        if ($contenttype !== '') {
+            $params['ResponseContentType'] = $contenttype;
+        }
+
+        if ($this->signingmethod == 'cf') {
```

Comment:
This method is a bit mixed with the old s3 and the new cf stuff, can you please move all the old s3 stuff to another method eg like

generate_presigned_url_cloudfront
generate_presigned_url_s3

and this method just swaps between them